workflow-runtime/src/commonMain/kotlin/com/squareup/workflow1/WorkflowInterceptor.kt

```diff
@@ -129,6 +142,21 @@ public interface WorkflowInterceptor {
     session: WorkflowSession
   ): Snapshot? = proceed(state)
 
+  /**
+   * Called when a [RenderingAndSnapshot] is updated from the runtime (i.e. when the [StateFlow]
+   * is provided a new value - this may still be de-duplicated).
```

Comment:
"this may still be de-duplicated" => by consumers of the state flow, based on some other things?

Reply:
by the `StateFlow` itself which de-duplicates emissions downstream.